internal/datastore/memdb/revisions.go

```diff
@@ -59,24 +59,42 @@ func (mdb *memdbDatastore) OptimizedRevision(ctx context.Context) (datastore.Rev
 }
 
 func (mdb *memdbDatastore) CheckRevision(ctx context.Context, revisionRaw datastore.Revision) error {
+	mdb.RLock()
+	defer mdb.RUnlock()
+
 	dr, ok := revisionRaw.(revision.Decimal)
 	if !ok {
 		return datastore.NewInvalidRevisionErr(revisionRaw, datastore.CouldNotDetermineRevision)
 	}
-	return mdb.checkRevisionLocal(dr)
+	return mdb.checkRevisionLocalNoLock(dr)
 }
 
-func (mdb *memdbDatastore) checkRevisionLocal(revisionRaw revision.Decimal) error {
+func (mdb *memdbDatastore) checkRevisionLocalNoLock(revisionRaw revision.Decimal) error {
```

Comment:
Maybe change this to checkRevisionLocalCallerLocks or something that tells the caller that the invariant must hold.

Reply:
Changed to `checkRevisionLocalCallerMustLock` to be as explicit as possible